usaspending_api/common/tests/unit/test_fiscal_year_helpers.py

```diff
@@ -69,41 +72,48 @@ def test_dates_are_fiscal_year_bookends():
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 2, 15)) == 1
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 3, 1)) == 1
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 4, 1)) == 1
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 5, 1)) == 1
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 5, 15)) == 1
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 5, 16)) == 2
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 6, 1)) == 2
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 7, 1)) == 2
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 8, 1)) == 2
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 9, 1)) == 3
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 10, 1)) == 3
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 11, 1)) == 3
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 12, 1)) == 4
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2001, 1, 1)) == 4
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2001, 2, 1)) == 4
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2001, 3, 1)) == 4
-
-    # Test a fiscal year WAY in the future.  We should definitely have flying cars by then.
-    assert fyh.calculate_last_completed_fiscal_quarter(2010, date(2001, 3, 1)) is None
-
-    # And one test using the default as_of_date just to make sure nothing catches on fire.
-    assert fyh.calculate_last_completed_fiscal_quarter(2000) == 4
+    now = datetime.now(timezone.utc)
+    yesterday = now + timedelta(days=-1)
+    tomorrow = now + timedelta(days=1)
+    current_fy = fyh.generate_fiscal_year(now)
+
+    mommy.make(
+        "submissions.DABSSubmissionWindowSchedule",
+        submission_fiscal_year=2000,
+        submission_reveal_date=now,
+        submission_fiscal_quarter=1,
+    )
+    mommy.make(
+        "submissions.DABSSubmissionWindowSchedule",
+        submission_fiscal_year=2010,
+        submission_reveal_date=tomorrow,
+        submission_fiscal_quarter=2,
+    )
+    mommy.make(
+        "submissions.DABSSubmissionWindowSchedule",
+        submission_fiscal_year=current_fy,
+        submission_reveal_date=yesterday,
+        submission_fiscal_quarter=3,
+    )
+    mommy.make(
+        "submissions.DABSSubmissionWindowSchedule",
+        submission_fiscal_year=current_fy,
+        submission_reveal_date=now,
+        submission_fiscal_quarter=4,
+    )
+    mommy.make(
+        "submissions.DABSSubmissionWindowSchedule",
+        submission_fiscal_year=current_fy,
+        submission_reveal_date=tomorrow,
+        submission_fiscal_quarter=5,
+    )
+
+    assert fyh.calculate_last_completed_fiscal_quarter(2000) == 1
```

Comment:
Would add another test case here to make sure that you get back Q1 through Q3 if they are completed fiscal quarters (kinda similar to the tests that are being removed).

Reply:
@sethstoudenmier I don't follow your thinking. The function only returns a single value for the latest quarter "revealed" (if any); you mean to add other test cases simulating different "now" dates? I think I have those covered below (last value is 4, not 3 or 5 though they are all in the same year).

Reply:
Followed up in Slack pertaining to this and the larger comment above. Outcome is to look at using the `is_quarter` column to avoid returning a quarter that has only a single period revealed.